### PR TITLE
feat: CatalogEntry json from nuspec

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #221 
Build CatalogEntry index.json item from package nuspec metadata.

### CODE REVIEW

File: src/main/java/com/artipie/nuget/metadata/CatalogEntry.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        private Map<String, List<Pair<String, String>>> dependenciesByTargetFramework() {
+            final Map<String, List<Pair<String, String>>> res = new HashMap<>();
+            this.nuspec.dependencies().forEach(
+                item -> {
+                    final String[] arr = item.split(":");
+                    final List<Pair<String, String>> dep;
+                    if (arr[0].isEmpty()) {
+                        dep = Collections.emptyList();
+                    } else {
+                        dep = new ArrayList<>(1);
```
Comment (dgarus) [inline src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:123]: Can we use `Collections.singletonList()`?

File: src/main/java/com/artipie/nuget/metadata/CatalogEntry.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    final JsonObjectBuilder item = Json.createObjectBuilder();
+                    if (!key.isEmpty()) {
+                        item.add("targetFramework", key);
+                    }
+                    final JsonArrayBuilder deps = Json.createArrayBuilder();
+                    val.forEach(
+                        pair -> {
+                            final JsonObjectBuilder dep = Json.createObjectBuilder();
+                            dep.add("id", pair.getKey());
+                            if (!pair.getValue().isEmpty()) {
```
Comment (dgarus) [inline src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:94]: Can `pair.getValue()` be `null`?

**Review by dgarus — CHANGES_REQUESTED** [review]
@olenagerasimova thanks!
There are a few questions.

File: src/main/java/com/artipie/nuget/metadata/CatalogEntry.java (hunk 2 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:94]: no, `null` is not possible (it follows from `dependenciesByTargetFramework` method implementation and the way `String.split()` function works), but it's not obligatory for nuget dependency to have version, so there can be empty line.

File: src/main/java/com/artipie/nuget/metadata/CatalogEntry.java (hunk 1 above)
Comment (dgarus) [reply src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:123]: In that case, capacity 1 will lead to extending List's array (create new array and copy all elements) in every merge.

Comment (olenagerasimova) [reply src/main/java/com/artipie/nuget/metadata/CatalogEntry.java]: sure, let's make it 10 - should be enough for most cases

Comment (dgarus) [reply src/main/java/com/artipie/nuget/metadata/CatalogEntry.java]: That is precisely what default is `java.util.ArrayList#DEFAULT_CAPACITY`

File: src/main/java/com/artipie/nuget/metadata/CatalogEntry.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        private Map<String, List<Pair<String, String>>> dependenciesByTargetFramework() {
+            final Map<String, List<Pair<String, String>>> res = new HashMap<>();
+            this.nuspec.dependencies().forEach(
+                item -> {
+                    final String[] arr = item.split(":");
+                    final List<Pair<String, String>> dep;
+                    if (arr[0].isEmpty()) {
+                        dep = Collections.emptyList();
+                    } else {
+                        dep = new ArrayList<>(10);
```
Comment (dgarus) [inline src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:124]: `10` is an extra parameter. 
`java.util.ArrayList#DEFAULT_CAPACITY = 10`

**Review by dgarus — CHANGES_REQUESTED** [review]
Minor comment

File: src/main/java/com/artipie/nuget/metadata/CatalogEntry.java (hunk 3 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:124]: hm, there seem to be some bug inside qulice, it complains, that

`ArrayList should be initialized with a size parameter (ConditionalRegexpMultilineCheck)`

and suppressing for this check is not working.

Comment (dgarus) [reply src/main/java/com/artipie/nuget/metadata/CatalogEntry.java:124]: ok, let's use `12`. Otherwise, it looks strange.

**Review by dgarus — APPROVED** [review]
Thanks!
LGTM

**Pull request merged**